libcloud/common/base.py

```diff
@@ -527,6 +527,7 @@ def request(
         stream=False,
         json=None,
         retry_failed=None,
+        container=None,
```

Comment:
Overall, the changes except this one look good to me.

I don't think we should add ``container`` argument this function since it's a generic function not tied to any specific API so we should create a dependency on the storage API / container argument here.

Reply:
Also, did you push all the changes? Since I don't see ``container`` argument you added in a bunch of places actually being used (or did I miss something)?

Reply:
The reason I changed this line is to avoid influencing other storage providers, for example, s3.
My previous changes were not a good idea because they added several lines into `libcloud/storage/base.py` so that other providers' upload functions will throw exceptions.
I have reverted all the changes that lead to this situation, and the new solution will only affect `libcloud/storage/drivers/oss.py`. The possible side effect is that if the method `_upload_object` of class `StorageDriver` in `libcloud/storage/base.py` changes, the OSS-related method also needs to be changed if necessary.

Reply:
Thanks - this new approach indeed looks much better and safer :)